shared/actions/fs/platform-specific.ios.js

```diff
@@ -1,15 +1,36 @@
 // @flow
 import * as Saga from '../../util/saga'
+import * as Types from '../../constants/types/fs'
 import * as FsGen from '../fs-gen'
 import {share, save} from './common.native'
+import {saveAttachmentDialog, showShareActionSheet} from '../platform-specific'
 
-function copyToDownloadDir(path: string, mime: string) {
-  return new Promise((resolve, reject) => resolve())
+function platformSpecificIntentEffect(
```

Comment:
I don't love the code separation appearing here. IMO the `saveAttachmentDialog`, `showShareActionSheet`, and `copyToDownloadDir` code should be where this differentiates, and this block here shouldn't be platform specific. It's too close to try to pick apart the differences.

Reply:
Nojima mentioned that `copyToDownloadDir` doesn't make sense on desktop, which motivated me to make this change.

Although I'm thinking separating this download saga into smaller chunks to help reusing code like in my loadMimeTypes PR. I'll see what I can do there to make it clearer.